### ISSUE TITLE
ZoL is missing the "multi_vdev_crash_dump" feature

### ISSUE THREAD

**Issue by FransUrbo** [zfsonlinux/zfs#2438]
Is apparently available in the Illumos code...

**Comment by darkfires** [comment]
Is there no way to ignore this feature for a read only recovery?  Have a few zpools that were created on illumos that I need to recover somehow...

**Comment by behlendorf** [comment]
@darkfires you should be able to import the pool read-only using the the latest master code.  If this doesn't work can you please report the exact error message your seeing.

**Comment by behlendorf** [comment]
Unfortunately, this functionality was implemented in a way which doesn't allow it to be read-only compatible.  Specifically the addition of the NOPARITY checksum type.

```
GUID	com.joyent:multi_vdev_crash_dump
READ-ONLY COMPATIBLE	no
DEPENDENCIES	none
```

**Comment by Thomas-1985** [comment]
Hello

Is there any progress in implementing this feature flag? I need it to import a ZFS pool that has been created with FreeNAS (9.3.1) on my linux box. The only feature flag that is missing is "multi_vdev_crash_dump" and i thing this would really enhance the cross-platform compatibility of ZFS

Best,
Thomas

**Comment by behlendorf** [comment]
@Thomas-1985 this hasn't been a priority for us since it's not required for Linux.  Although I agree it would be nice to have solely for improved compatibility.

**Comment by satmandu** [comment]
FYI for what it is worth, the most recent stable build of openzfsonosx appears to have this feature flag as an option:

```
sudo zpool status
  pool: externalzpool
 state: ONLINE
status: Some supported features are not enabled on the pool. The pool can
	still be used, but some features are unavailable.
action: Enable all features using 'zpool upgrade'. Once this is done,
	the pool may no longer be accessible by software that does not support
	the features. See zpool-features(5) for details.
  scan: scrub in progress since Sun May  1 15:31:49 2016

sudo zpool upgrade
This system supports ZFS pool feature flags.

All pools are formatted using feature flags.


Some supported features are not enabled on the following pools. Once a
feature is enabled the pool may become incompatible with software
that does not support the feature. See zpool-features(5) for details.

POOL  FEATURE
---------------
externalzpool
      multi_vdev_crash_dump
      sha512
      skein
      edonr

```

(I haven't upgraded to support those feature flags yet for fear that I won't be able to access this zpool from ZoL, but I imagine new zpools from OS X with a stable openzfsonosx build will have those feature flags enabled by default.)

**Comment by darkfires** [comment]
FreeBSD supports the flags as well.  I actually completely moved all of my ZFS systems to FreeBSD from ZoL because of too many problems and performance issues, also having ZFS natively integrated into the OS makes life so much easier when installing updates.  Small things like ZFS's ARC stats included into top just add to the polish.  Honestly wish I had done it way sooner than I did.  No systemd on FreeBSD is also nice :)

**Comment by maaku** [comment]
I am now in a position of having my data stuck because of this issue.

For unrelated, irrelevant reasons I'm forced to move a server which has 72TB of storage from FreeBSD (FreeNAS) to GNU/Linux (Ubuntu). The only holdup is that `multi_vdev_crash_dump` is enabled on the pool.

I do not have a 2nd 72TB storage pool just lying around to do a send & receive from. So the only option seems to be adding support for this flag (or better yet, a tool to remove this flag as the feature was never once used by the pool). Any pointers for how best to do this? I'm not familiar with the ZFS codebase or ZoL so some indication of where to look or how complicated a task this is would be appreciated.

**Comment by DeHackEd** [comment]
While the implementation is fairly straight-forward, most of the changes are in files that are OS-specific. The vdev_disk and ZVOL drivers are all linux-specific and are where most of the modifications are. It makes porting less straight-forward than the other features since they must be re-implemented.

**Comment by behlendorf** [comment]
>  (or better yet, a tool to remove this flag as the feature was never once used by the pool)

@maaku I'd suggest taking a look at PR #3878, it extends the `zhack` command to allow you to force disable a feature flag.  Now keep in mind that this is a developer tool and **in general this not a safe thing to do!**  However, in the case of the `multi_vdev_crash_dump` clearing this feature flag should be safe and it will allow you to import the pool.

**Comment by DeHackEd** [comment]
Late to the party, 42db43e982a243342a38e13f513513bb164d45e3 should close this issue.

**Comment by gmelikov** [comment]
@DeHackEd IIRC this commit is a dummy to support pools with this feature, but I think that we can close this issue.

**Comment by DeHackEd** [comment]
Well, yes. But the pool feature mainly introduces a new dataset property `checksum=noparity` which is "supported" (3c67d83a8afb391f20bc53d36a0cebea6897b3e2) weakly from an older commit. This was just getting the feature flag explicitly listed which was the import blocker. Between the two of them Linux can manage with a pool imported from other systems just fine. It just can't read/write to such datasets, which is arguably of little use on Linux anyway since they are crash dumps from different operating systems.
